gpbft/src/payload.rs

```diff
@@ -0,0 +1,266 @@
+// Copyright 2019-2024 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use crate::chain::{Cid, ECChain};
+use strum_macros::Display;
+
+/// Fields of the message that make up the signature payload in the GPBFT consensus protocol
+#[derive(PartialEq, Eq)]
+pub struct Payload {
+    /// GPBFT instance number
+    pub instance: u64,
+    /// GPBFT round number
+    pub round: u64,
+    /// Current phase of the GPBFT protocol
+    pub phase: Phase,
+    /// Additional data related to this consensus instance
+    pub supplemental_data: SupplementalData,
+    /// The agreed-upon value for this instance
+    pub value: ECChain,
+}
+
+impl Payload {
+    /// Creates a new Payload instance
+    ///
+    /// # Arguments
+    ///
+    /// * `instance` - The GPBFT instance number
+    /// * `round` - The current round number
+    /// * `phase` - The current phase of the protocol
+    /// * `supplemental_data` - Additional data for this instance
+    /// * `value` - The agreed-upon ECChain
+    pub fn new(
+        instance: u64,
+        round: u64,
+        phase: Phase,
+        supplemental_data: SupplementalData,
+        value: ECChain,
+    ) -> Self {
+        Payload {
+            instance,
+            round,
+            phase,
+            supplemental_data,
+            value,
+        }
+    }
+
+    /// Serializes the payload for signing.
+    /// Format: `GPBFT:network_name:phase+round+instance+commitments+chain_key+power_table_cid`
+    pub fn serialize_for_signing(&self, network_name: &str) -> Vec<u8> {
```

Comment:
I'd like to have some `go-f3` interop tests for such functions, but maybe we could tackle that in a subsequent PR.